test/e2e-tests/support/kustomize.go

```diff
@@ -0,0 +1,111 @@
+package support
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	pipepinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func ResourceToType[T any](resource *resource.Resource, t *T) error {
+	bytes, err := resource.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(bytes, &t)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func KustomizeBuild(path string) (resmap.ResMap, error) {
+	options := krusty.MakeDefaultOptions()
+	k := krusty.MakeKustomizer(options)
+	fs := filesys.FileSystemOrOnDisk{
+		FileSystem: nil,
+	}
+
+	resourceMap, err := k.Run(fs, path)
+	if err != nil {
+		return nil, err
+	}
+
+	return resourceMap, nil
+}
+
+func CreateObjectsFromResourceMap(ctx context.Context, clients *Clients, resourceMap resmap.ResMap) error {
+	for _, rsc := range resourceMap.Resources() {
+		kind := rsc.GetKind()
+		switch kind {
+		case "ConfigMap":
+			{
+				var configMap corev1.ConfigMap
+				err := ResourceToType(rsc, &configMap)
+				if err != nil {
+					return err
+				}
+
+				_, err = clients.Kubernetes.CoreV1().ConfigMaps("test-namespace").Get(ctx, configMap.Namespace, metav1.GetOptions{})
+				if err == nil {
+					return nil
+				}
+
+				_, err = clients.Kubernetes.CoreV1().ConfigMaps("test-namespace").Create(ctx, &configMap, metav1.CreateOptions{})
```

Comment:
It looks like there's an implicit assumption here that the namespace is called `test-namespace`, but that is no longer being created by the Make recipe, right? Should/could this be changed to not rely on the namespace name being `test-namespace`?

Reply:
I tried to find a way to get the current active namespace that is set in the kubeconfig from oc but couldn't figure it out. I created a namespace parameter for the tests instead